crates/revm/revm-primitives/src/env.rs

```diff
@@ -176,6 +176,7 @@ where
         Transaction::Deposit(TxDeposit { to, mint, value, gas_limit, input, .. }) => {
             tx_env.gas_limit = *gas_limit;
             if let Some(m) = mint {
+                // TODO: why is this not zero?
                 tx_env.gas_price = U256::from(*m);
```

Comment:
@clabby Shouldn't this be always 0 for deposits? Or is this a gas accounting trick in order to make the refunds work out that I don't understand?